src/gui/search/searchwidget.cpp

```diff
@@ -381,3 +388,9 @@ void SearchWidget::closeTab(int index)
 
     delete tab;
 }
+
+void SearchWidget::closeAllTabs()
+{
+    while (m_allTabs.size())
```

Comment:
Better use a for loop to cut down all those function calls: `for (int i = (m_allTabs.size() - 1); i >= 0; --i)`